### PR TITLE
Update graph API return codes.

### PR DESCRIPTION
**Pull request by hidmic**
Connected to https://github.com/ros2/rmw/pull/272. This patch affects:
- `rmw_get_node_names()`
- `rmw_get_node_names_with_enclaves()`
- `rmw_get_topic_names_and_types()`
- `rmw_get_service_names_and_types()`
- `rmw_get_publishers_info_by_topic()`
- `rmw_get_subscriptions_info_by_topic()`
- `rmw_get_subscriber_names_and_types_by_node()`
- `rmw_get_publisher_names_and_types_by_node()`
- `rmw_get_service_names_and_types_by_node()`
- `rmw_get_client_names_and_types_by_node()`
- `rmw_count_publishers()`
- `rmw_count_subscribers()`

### CODE REVIEW

File: rmw_connext_shared_cpp/src/node_names.cpp [hunk 1]
```diff
@@ -35,19 +36,17 @@ get_node_names_impl(
...
-    return RMW_RET_ERROR;
+  RMW_CHECK_ARGUMENT_FOR_NULL(node, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    node,
+    node->implementation_identifier,
+    implementation_identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+  if (RMW_RET_OK != rmw_check_zero_rmw_string_array(node_names)) {
+    return RMW_RET_INVALID_ARGUMENT;
   }
```
Comment (ivanpauno) [inline rmw_connext_shared_cpp/src/node_names.cpp:47]: this seems inconsistent with:
```cpp
  ret = rmw_names_and_types_check_zero(topic_names_and_types);
  if (RMW_RET_OK != ret) {
    return ret;
  }
```

Comment (Lobotuerk) [reply rmw_connext_shared_cpp/src/node_names.cpp:47]: My guess is that `rmw_check_zero_rmw_string_array(node_names)` does not return `RMW_RET_INVALID_ARGUMENT`

Comment (ivanpauno) [reply rmw_connext_shared_cpp/src/node_names.cpp:47]: ok, it's a bit weird the semantic different between the two, but it doesn't matter much,

**Review by ivanpauno — APPROVED** [review]

**Review by Lobotuerk — APPROVED** [review]
LGTM

File: rmw_connext_shared_cpp/src/node_names.cpp (hunk 1 above)
Comment (hidmic) [reply rmw_connext_shared_cpp/src/node_names.cpp:47]: Indeed. `rmw_check_zero_rmw_string_array` returns `RMW_RET_INVALID_ARGUMENT` if the argument is NULL, and `RMW_RET_ERROR` if it's not zero initialized. We can change `rmw_names_and_types_check_zero` call sites to the do the same for consistency's sake though.

**Comment by hidmic** [comment]
CI up to `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12375/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7344/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10087/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12287/)

**Comment by hidmic** [comment]
CI up to `test_rmw_implementation`, `rcl`, and `rclcpp`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12388/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7356/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10099/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12300/)

**Comment by hidmic** [comment]
Alright, going in! Thanks for all the good reviews!

**Pull request merged**